ethicml/utility/data_helpers.py

```diff
@@ -1,9 +1,34 @@
 """Helper functions for working with DataFrames (and Series)."""
+from typing import Optional, Union, overload
+
 import pandas as pd
 
-__all__ = ["shuffle_df"]
+__all__ = ["undo_one_hot", "shuffle_df"]
 
 
 def shuffle_df(df: pd.DataFrame, random_state: int) -> pd.DataFrame:
     """Shuffle a given dataframe."""
     return df.sample(frac=1.0, random_state=random_state).reset_index(drop=True)
+
+
+@overload
+def undo_one_hot(df: pd.DataFrame, new_column_name: None = None) -> pd.Series:
+    ...
+
+
+@overload
+def undo_one_hot(df: pd.DataFrame, new_column_name: str) -> pd.DataFrame:
+    ...
+
+
+def undo_one_hot(
+    df: pd.DataFrame, new_column_name: Optional[str] = None
+) -> Union[pd.Series, pd.DataFrame]:
+    """Undo one-hot encoding."""
+    # we have to overwrite the column names because `idxmax` uses the column names
+    df.columns = pd.Index(list(range(df.shape[1])))
+    result = df.idxmax(axis="columns")
+    if new_column_name is not None:
+        return result.to_frame(name=new_column_name)
+    else:
+        return result
```

Comment:
Do the ellipsis serve some purpose? Or can they be replaced with a `raise NotImplementedError()`?

Reply:
you mean the ellipses from the overloads? I think it can also be `pass`. not sure about `NotImplementedError`.

Reply:
oh - I think I got myself confused. I thought they were saying if called with this function signature do something different. I didn't realise that they were there just for mypy